polyfill/test/validStrings.mjs

```diff
@@ -284,7 +284,12 @@ const calendarName = withCode(choice(...calendarNames), (data, result) => {
   if (!data.calendar) data.calendar = result;
 });
 const calendarAnnotation = seq('[', [annotationCriticalFlag], 'u-ca=', calendarName, ']');
-const annotations = oneOrMore(choice(calendarAnnotation, annotation));
+const annotations = withSyntaxConstraints(oneOrMore(choice(calendarAnnotation, annotation)), (result) => {
+  const numCalendarAnnotations = (result.match(/u-ca=/g) ?? []).length;
+  if (numCalendarAnnotations > 1 && /\[!u-ca=/.test(result)) {
+    throw new SyntaxError('more than one calendar annotation and at least one critical');
```

Comment:
What are the criteria for using SyntaxError vs. RangeError for cases where strings are invalid? Are we currently throwing SyntaxError in all of the places we should be?

Reply:
SyntaxError is just a quirk of this particular test setup in `validStrings.mjs`. We don't throw SyntaxError anywhere in Temporal.